tests/v2/test_keyfile.nim

```diff
@@ -0,0 +1,119 @@
+{.used.}
+
+import
+  std/[json, os],
+  testutils/unittests, chronos, chronicles,
+  eth/keys,
+  ../../waku/v2/utils/keyfile,
+  ../test_helpers
+
+from ../../waku/v2/protocol/waku_noise/noise_utils import randomSeqByte
+
+suite "KeyFile test suite":
+
+  let rng = newRng()
+
+  test "Create/Save/Load single keyfile":
+
+    let password = "randompassword"
+    let filepath = "./test.keyfile"
+
+    var secret = randomSeqByte(rng[], 300)
+    let keyfile = createKeyFileJson(secret, password)
+
+    check:
+      keyfile.isOk()
+      saveKeyFile(filepath, keyfile.get()).isOk()
+
+    var decodedSecret = loadKeyFile("test.keyfile", password)
+
+    check:
+      decodedSecret.isOk()
+      secret == decodedSecret.get()
+
+    removeFile(filepath)
```

Comment:
As a nitpick, aren't we risking here that `removeFile` is not executed if an exception is thrown before? i.e. if `loadKeyFile` raises an exception, the file will never be removed.

I would use `defer` at the begining:

```
test "Create/Save/Load single keyfile":

    let password = "randompassword"
    let filepath = "./test.keyfile"

    defer: removeFile(filepath)

    ...
```

Reply:
I can see nimbus extensively uses it in testing for [closing files](https://github.com/status-im/nimbus-eth2/blob/v22.10.1/tests/test_keystore_management.nim#L42) and [closing databases](https://github.com/status-im/nimbus-eth2/blob/v22.10.1/tests/slashing_protection/test_slashing_protection_db.nim#L56) and in production code for [removing secrets](https://github.com/status-im/nimbus-eth2/blob/v22.10.1/beacon_chain/spec/keystore.nim#L804) and [closing files](https://github.com/status-im/nimbus-eth2/blob/v22.10.1/beacon_chain/validators/keystore_management.nim#L411). Would be a pitty if Nim removes it, but I don't see any immediate plans of doing it.

imho `try` `finally` adds an extra ugly indentation level and `defer` solves the same issue more elegantly. Nimbus using it gives me confidence, so its why I would prefer `defer`, but feel free to use `finally`, it also solves the problem I pointed out of the file not being removed :)

Reply:
ow nice reference! i actually didn't know it was translated to a try/finally.
resolving thanks!